R/layers.R

```diff
@@ -120,6 +120,31 @@ sass_layer <- function(
   )
 }
 
+
+#' @export
+#' @param file file path to a `.scss` file.
+#' @describeIn sass_layer Read in a `.scss` file with parse special `/*-- scss:(functions|defaults|rules|mixins) --*/` comments as relevant sections of a `sass_layer()`.
+sass_layer_file <- function(file) {
+  src <- readLines(file)
+  # https://github.com/quarto-dev/quarto-cli/blob/3d6063/src/command/render/sass.ts#L119-L123
+  pattern_key <- "scss:(functions|defaults|rules|mixins)"
+  pattern <- paste0("^/\\*--[ \\t]*", pattern_key, "[ \\t]*--\\*/$")
+  idx <- grep(pattern, src)
+  if (!any(idx)) {
```

Comment:
Tiny nit: calling `any(idx)` is slightly confusing because it's an integer vector of indices of matching elements, not a logical vector indicating which ones match.

Using `any(idx)` would make more sense if `grepl()` were used above. Or this check could be `if (length(idx) == 0)`.

Reply:
Oh whoops, yea I was using `grepl()` at first and forgot to update this check, thanks!